File: src/系统/03．技能系统/04．快捷键技能/00．配置ID工具.lua
local ____lualib = require("lualib_bundle")
local __TS__StringSubstring = ____lualib.__TS__StringSubstring
local __TS__StringTrim = ____lualib.__TS__StringTrim
local ____exports = {}
local ____require_result_0 = require("lib.扩展函数.封装函数.01．通用工具.01．FourCC转换安全版")
local stringToFourCCSafe = ____require_result_0.stringToFourCCSafe
local _____914D_7F6EID_7F13_5B58 = {}
--- 读取“名称#四位内部ID”；没有名称时也兼容直接传入四位 Rawcode。
____exports["解析配置内部ID"] = function(_____914D_7F6E_503C)
    if _____914D_7F6E_503C == nil or _____914D_7F6E_503C == "" then
        return 0
    end
    local _____5DF2_7F13_5B58 = _____914D_7F6EID_7F13_5B58[_____914D_7F6E_503C]
    if _____5DF2_7F13_5B58 ~= nil then
        return _____5DF2_7F13_5B58
    end
    local _____5206_9694_4F4D_7F6E = (string.find(_____914D_7F6E_503C, "#", nil, true) or 0) - 1
    local _____539F_59CBID = _____5206_9694_4F4D_7F6E > 0 and __TS__StringTrim(__TS__StringSubstring(_____914D_7F6E_503C, _____5206_9694_4F4D_7F6E + 1)) or __TS__StringTrim(_____914D_7F6E_503C)
    if #_____539F_59CBID ~= 4 then
        _____914D_7F6EID_7F13_5B58[_____914D_7F6E_503C] = 0
        return 0
    end
    local _____7C7B_578BID = stringToFourCCSafe(_____539F_59CBID)
    _____914D_7F6EID_7F13_5B58[_____914D_7F6E_503C] = _____7C7B_578BID
    return _____7C7B_578BID
end
____exports["解析配置内部ID列表"] = function(_____914D_7F6E_5217_8868)
    local _____7ED3_679C = {}
    do
        local i = 0
        while i < #_____914D_7F6E_5217_8868 do
            local _____7C7B_578BID = ____exports["解析配置内部ID"](_____914D_7F6E_5217_8868[i + 1])
            if _____7C7B_578BID > 0 then
                _____7ED3_679C[#_____7ED3_679C + 1] = _____7C7B_578BID
            end
            i = i + 1
        end
    end
    return _____7ED3_679C
end
return ____exports

File: TS/系统/03．技能系统/04．快捷键技能/00．配置ID工具.ts
/** @noSelfInFile */

const { stringToFourCCSafe } = require("lib.扩展函数.封装函数.01．通用工具.01．FourCC转换安全版") as {
  stringToFourCCSafe: (this: void, rawId: string | undefined | null) => number;
};

const 配置ID缓存: Record<string, number | undefined> = {};

/** 读取“名称#四位内部ID”；没有名称时也兼容直接传入四位 Rawcode。 */
export function 解析配置内部ID(this: void, 配置值: string | undefined | null): number {
  if (配置值 == null || 配置值 === "") return 0;

  const 已缓存 = 配置ID缓存[配置值];
  if (已缓存 != null) return 已缓存;

  const 分隔位置 = 配置值.indexOf("#");
  const 原始ID = 分隔位置 > 0 ? 配置值.substring(分隔位置 + 1).trim() : 配置值.trim();
  if (原始ID.length !== 4) {
    配置ID缓存[配置值] = 0;
    return 0;
  }

  const 类型ID = stringToFourCCSafe(原始ID);
  配置ID缓存[配置值] = 类型ID;
  return 类型ID;
}

export function 解析配置内部ID列表(this: void, 配置列表: readonly (string | undefined | null)[]): number[] {
  const 结果: number[] = [];
  for (let i = 0; i < 配置列表.length; i++) {
    const 类型ID = 解析配置内部ID(配置列表[i]);
    if (类型ID > 0) 结果.push(类型ID);
  }
  return 结果;
}

export {};
